homework1/task05/maximal_subarray.py

```diff
@@ -0,0 +1,31 @@
+"""
+Given a list of integers numbers "nums".
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:

```

Comment:
1) use slices
2) sum of all elements in array can be computed by using function sum()
3) finding max element in a list can be used by utilizing function max()

When you are writing programs in python, if you have to use indexes and multiple nested loops, usually it means you are doing something wrong.